react/features/base/devices/functions.js

```diff
@@ -1,8 +1,33 @@
 // @flow
 
+import { parseURLParams } from '../config';
 import JitsiMeetJS from '../lib-jitsi-meet';
 import { updateSettings } from '../settings';
 
+declare var APP: Object;
+
+/**
+ * Detects the use case when the labels are not available if the A/V permissions
+ * are not yet granted.
+ *
+ * @param {Object} state - The redux state.
+ * @returns {boolean} - True if the labels are already initialized and false
+ * otherwise.
+ */
+export function areDeviceLabelsInitialized(state: Object) {
+    if (APP.conference._localTracksInitialized) {
```

Comment:
You don't want to move this value into redux? It feels like it would be pretty straightforward.

Reply:
Yes it will be easy. I was thinking about moving it but at the end it I decided to leave it as it is. It is very tied and specific to the whole conference.js flow and logic and it is mainly used there. Once we refactor the conference.js I bet there will be a better way of checking this.

Reply:
When is the conference refactor scheduled for?

Reply:
I guess @yanas can answer this...